Engine/GameObject.cs

```diff
@@ -1,126 +1,158 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Collections.ObjectModel;
-
 using Engine.BaseAssets.Components;
 
 namespace Engine
 {
-    public sealed class GameObject : SerializedObject
+    public sealed class GameObject : SerializableObject
     {
-        private bool enabled = true;
-        public bool Enabled { get => Transform.Parent == null ? enabled : enabled && Transform.Parent.GameObject.enabled; set => enabled = value; }
-        public Transform Transform { get; private set; }
-        private List<Component> components = new List<Component>();
+        [SerializedField]
+        private string name;
+        [SerializedField]
+        private bool localEnable = true;
+        [SerializedField]
+        private Transform transform = null;
+        [SerializedField]
+        private readonly List<Component> components = new List<Component>();
+
+        public string Name { get => name; set => name = value; }
```

Comment:
make sure name is not null or empty/whitespace

Reply:
Done